### DISCUSSION TITLE
Event handler not acting on events from taskReferenceName with iteration appended

### DISCUSSION THREAD

**Discussion by mpsanchezg** [Netflix/conductor#2379] [Q&A] [answered]
Hi, I have the following problem.
Inside a WHILE DO I'm triggering events with an Event Task. The problem is that an iteration number is appended to that task's taskReferenceName. This causes my Event Handler not to handle those events (since the name/location doesn't match).

Is this a bug? How can I accomplish this?

**Comment by kishorebanala** [comment, marked as answer]
Hey @mpsanchezg, before we categorize this as a bug, we'd like to better understand your use case. Can you provide more details about how you are tying these events in WHILE DO to Event handler, and what you're trying to accomplish please?

Event handler expects `taskReferenceName` in event payload. Would it be possible for you to send this in the event payload instead of dynamic taskReferenceName generated by WHILE DO?

**Reply by mpsanchezg** [reply]
Hi,
This is the EVENT task that is within the WHILE DO task:
```
{
  "type": "EVENT",
  "name": "start_event",
  "taskReferenceName": "start_event",
  "inputParameters": {
    "sourceWorkflowId": "${wf.output.result}"
  },
  "sink": "conductor",
  "asyncComplete": false
}
```
And the event handler:
```
{
  "name": "start_eventHandler",
  "event": "conductor:workflow_name:start_event",
  "actions": [
    {
      "action": "complete_task",
      "complete_task": {
        "workflowId": "${sourceWorkflowId}",
        "taskRefName": "wait"
      }
    }
  ],
  "active": true
}
```
But, in WHILE DO output:
``` 
"1": {
      "wf": {
         "result": "asdfasf"
      },
      "start_event": {
         "event_produced": "conductor:workflow_name:start_event__1",
         "sourceWorkflowId": "asdfasf",
         "asyncComplete": false,
         "sink": "conductor",
         "workflowType": "workflow_name",
         "correlationId": null,
         "workflowVersion": 1,
         "workflowInstanceId": "qwerewrwetr"
      }
   }, ...
``` 
The event produced is not the event that I defined, because of the `__i` that occurs in all iterations, `__1` in this case.
Are you suggesting to add "taskReferenceName" in the "inputParameters"?

**Reply by kishorebanala** [reply]
@mpsanchezg If you look at https://github.com/Netflix/conductor/blob/main/core/src/main/java/com/netflix/conductor/core/execution/tasks/Event.java#L118, using `conductor:my_queue_name` in sink (instead of `"sink": "conductor",` in Event task config) would then send events to 

```
"conductor:" + workflow.getWorkflowName() + ":" + my_queue_name
```

which wouldn't use taskReferenceName at all.

**Reply by mpsanchezg** [reply]
THANKS

**Reply by mpsanchezg** [reply]
https://github.com/Netflix/conductor/pull/2393 
I made a PR adding this to the documentation.
